askama_derive/src/generator.rs

```diff
@@ -16,7 +17,9 @@ pub(crate) fn derive_template(input: TokenStream) -> TokenStream {
     let ast: syn::DeriveInput = syn::parse(input).unwrap();
     match build_template(&ast) {
         Ok(source) => source.parse().unwrap(),
-        Err(e) => e.into_compile_error(),
+        Err(e) => syn::Error::new(Span::call_site(), e)
+            .to_compile_error()
+            .into(),
```

Comment:
At this point I guess we could implement `to_compile_error()` manually (untested):

```rust
format!("::core::compile_error!({:?});", format_args!("{e}")).parse()
```

If we can omit  the `proc_macro2` dependency altogether in this case.